airbyte-integrations/connectors/source-twilio/source_twilio/streams.py

```diff
@@ -0,0 +1,376 @@
+
+    List of available phone number countries, as well as local, mobile and toll free numbers
+    may be different on each request, so could not pass the full refresh tests.
+    """
+
+    parent_stream = Accounts
+    data_field = "countries"
+    subresource_uri_key = "available_phone_numbers"
+    primary_key = None
+
+
+class AvailablePhoneNumbersLocal(TwilioNestedStream):
+    """https://www.twilio.com/docs/phone-numbers/api/availablephonenumberlocal-resource#read-multiple-availablephonenumberlocal-resources"""
+
+    parent_stream = AvailablePhoneNumberCountries
+    data_field = "available_phone_numbers"
+    subresource_uri_key = "local"
+    primary_key = None
+
+
+class AvailablePhoneNumbersMobile(TwilioNestedStream):
+    """https://www.twilio.com/docs/phone-numbers/api/availablephonenumber-mobile-resource#read-multiple-availablephonenumbermobile-resources"""
+
+    parent_stream = AvailablePhoneNumberCountries
+    data_field = "available_phone_numbers"
+    subresource_uri_key = "mobile"
+    primary_key = None
+
+
+class AvailablePhoneNumbersTollFree(TwilioNestedStream):
+    """https://www.twilio.com/docs/phone-numbers/api/availablephonenumber-tollfree-resource#read-multiple-availablephonenumbertollfree-resources"""
+
+    parent_stream = AvailablePhoneNumberCountries
+    data_field = "available_phone_numbers"
+    subresource_uri_key = "toll_free"
+    primary_key = None
+
+
+class IncomingPhoneNumbers(TwilioNestedStream):
+    """https://www.twilio.com/docs/phone-numbers/api/incomingphonenumber-resource#read-multiple-incomingphonenumber-resources"""
+
+    parent_stream = Accounts
+
+
+class Keys(TwilioNestedStream):
+    """https://www.twilio.com/docs/usage/api/keys#read-a-key-resource"""
+
+    parent_stream = Accounts
+
+
+class Calls(TwilioNestedStream, IncrementalTwilioStream):
+    """https://www.twilio.com/docs/voice/api/call-resource#create-a-call-resource"""
+
+    parent_stream = Accounts
+    incremental_filter_field = "EndTime>"
+    cursor_field = "end_time"
+
+
+class Conferences(TwilioNestedStream, IncrementalTwilioStream):
+    """https://www.twilio.com/docs/voice/api/conference-resource#read-multiple-conference-resources"""
+
+    parent_stream = Accounts
+    incremental_filter_field = "DateUpdated>"
+    time_filter_template = "%Y-%m-%d"
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        records = response.json().get(self.data_field, [])
+        if stream_state.get(self.cursor_field):
+            for record in records:
+                if pendulum.parse(record[self.cursor_field], strict=False) <= pendulum.parse(stream_state[self.cursor_field], strict=False):
+                    yield record
+        yield from records
+
+
+class ConferenceParticipants(TwilioNestedStream):
```

Comment:
question for my understanding: 

here we have a resource that doesn't support incremental but whose parent stream supports incremental. what is the behaviour we expect here? is `ConferenceParticipants` suppose to make retrieve all the `conference` objects and make requests for each `conference` item it receives?

Reply:
can we document this behaviour as docstrings maybe on the `IncrementalTwilioStream`? I want to make this as easy as possible on the next reader.

Reply:
Yes, the expected behavior is exactly as you described. Moreover, we make the full refresh read of parent objects all the times, even if both parent and child streams support incremental.
I've added a docstring describing that to `TwilioNestedStream` as it seem to be the right place to put it.

See [streams.py](https://github.com/airbytehq/airbyte/pull/4070/files/39277c140aba832724a6b5378fbb5ff803280cb5#diff-44937f7f5318fa78801189e983a9d2c1ace98936482724748f52d79af920938dR142-R156) for reference.